### DISCUSSION TITLE
Provide new function in Tools class to get current URL of the page.

### DISCUSSION THREAD

**Discussion by Amit-Kumar-Tiwari-Webkul** [PrestaShop/PrestaShop#33092] [Ideas]
### The problem

Suppose, we have created a controller page with several customs parameters and want to assign a page URL to JS.
If the page reloaded then we get only the base URL of the page with the token.
So our suggestion is, there must be a function by which we can get the Current page URL of any page either back office or front office

### The solution

We can provide a function for this.

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [ ] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
Hello @Amit-Kumar-Tiwari-Webkul 👍🏻 

There's `Tools::getCurrentUrl()`. It is available since PrestaShop 8.1.

**Comment by Amit-Kumar-Tiwari-Webkul** [comment]
Hello,
Thanks for the information.
